### PR TITLE
OnStop improvements

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

**Comment by stgraber** [comment]
Note that I could quite kill the goroutine entirely. The reason is that since the monitor is still single threaded, nothing within OnStop can use it as it would cause a deadlock.

So we still need to let OnStop itself return, then the goroutine can do its job and clean things up and LXD itself can check for its status through the operation map.

File: lxd/container_lxc.go [hunk 1]
```diff
@@ -1852,10 +1836,7 @@ func (c *containerLXC) OnStop(target string) error {
 		deviceTaskSchedulerTrigger("container", c.name, "stopped")
 
 		// Record current state
-		err = dbContainerSetState(c.daemon.db, c.id, "STOPPED")
-		if err != nil {
-			return
-		}
+		dbContainerSetState(c.daemon.db, c.id, "STOPPED")
```
Comment (tych0) [inline lxd/container_lxc.go]: Are we sure we want to ignore the error here? Should we at least log it?

**Comment by tych0** [comment]
Cool, thanks for looking into this!

File: lxd/container_lxc.go (hunk 1 above)
Comment (stgraber) [inline lxd/container_lxc.go]: Logging would probably be fine, returning on error was actually a bug as the error wouldn't be reported anywhere and if the container was ephemeral, we'd fail to delete it.

**Comment by stgraber** [comment]
There, that extra commit should take care of this. Both logging more errors and also, sending the error as part of the operation status (well, the last error we run into anyway).

**Pull request merged**